core/src/main/java/com/google/instrumentation/trace/PropagationUtil.java

```diff
@@ -0,0 +1,230 @@
+ *
+ * <pre>{@code
+ * private static final Tracer tracer = Tracer.getTracer();
+ * void onRequestReceived() {
+ *   String headerName = PropagationUtil.HTTP_HEADER_NAME;
+ *   SpanContext spanContext = PropagationUtil.fromHttpHeaderValue(headers.find(headerName));
+ *   try (NonThrowingCloseable ss =
+ *            tracer.spanBuilderWithRemoteParent(spanContext, "Recv.MyRequest").startScopedSpan() {
+ *     // Handle request and send response back.
+ *   }
+ * }
+ * }</pre>
+ */
+public final class PropagationUtil {
+  // Mask to extract a byte value.
+  private static volatile Handler handler = DefaultHandler.INSTANCE;
+
+  /** The header name that must be used in the HTTP request for the tracing context. */
+  public static final String HTTP_HEADER_NAME = "Trace-Context";
+
+  /**
+   * Serializes a {@link SpanContext} using the HTTP standard format.
+   *
+   * @param spanContext the {@code SpanContext} to serialize.
+   * @return the serialized US-ASCII encoded HTTP header value.
+   * @throws NullPointerException if the {@code spanContext} is null.
+   */
+  public static String toHttpHeaderValue(SpanContext spanContext) {
+    return BaseEncoding.base16().encode(toBinaryValue(spanContext));
+  }
+
+  /**
+   * Parses the {@link SpanContext} from the HTTP standard format.
+   *
+   * @param input a US-ASCII encoded buffer of characters from which the {@code SpanContext} will be
+   *     parsed.
+   * @return the parsed {@code SpanContext}.
+   * @throws NullPointerException if the {@code input} is null.
+   * @throws IllegalArgumentException if the {@code input} is invalid.
+   * @throws IllegalStateException if the version is not supported.
+   */
+  public static SpanContext fromHttpHeaderValue(CharSequence input) {
+    return fromBinaryValue(BaseEncoding.base16().decode(input));
+  }
+
+  /**
+   * Serializes a {@link SpanContext} using the binary format.
+   *
+   * @param spanContext the {@code SpanContext} to serialize.
+   * @return the serialized binary value.
+   * @throws NullPointerException if the {@code spanContext} is null.
+   */
+  public static byte[] toBinaryValue(SpanContext spanContext) {
+    checkNotNull(spanContext, "spanContext");
+    return handler.toBinaryFormat(spanContext);
+  }
+
+  /**
+   * Parses the {@link SpanContext} from the binary format.
+   *
+   * @param bytes a binary encoded buffer from which the {@code SpanContext} will be parsed.
+   * @return the parsed {@code SpanContext}.
+   * @throws NullPointerException if the {@code input} is null.
+   * @throws IllegalArgumentException if the {@code input} is invalid.
+   * @throws IllegalStateException if the version is not supported.
+   */
+  public static SpanContext fromBinaryValue(byte[] bytes) {
+    checkNotNull(bytes, "bytes");
+    return handler.fromBinaryFormat(bytes);
+  }
+
+  /**
+   * Sets a new {@link Handler}.
+   *
+   * @param newHandler the new {@code Handler} to be set.
+   */
+  public static void setHandler(Handler newHandler) {
```

Comment:
When will this method be called?  Would it be better to make the Handler an argument to the serialization and deserialization methods?